File: python/7.py
def prime_factors(n):
    i = 2

    factors = []

    while i * i <= n:
        if n % i == 0:
            n //= i
            factors.append(i)
        else:
            i += 1

    if n > 1:
        factors.append(n)

    return factors


def is_prime(n):
    return len(prime_factors(n)) == 1


primes = []
i = 0

while len(primes) < 10001:
    if is_prime(i):
        primes.append(i)

    i += 1

print(primes[-1])

File: swift/7.swift
func primeFactors(_ n: Int) -> [Int] {
    var i = n
    var j = 2

    var factors: [Int] = []

    while j * j <= i {
        if i % j == 0 {
            i /= j
            factors.append(j)
        } else {
            j += 1
        }
    }

    if i > 1 {
        factors.append(i)
    }

    return factors
}

func isPrime(_ n: Int) -> Bool {
    return primeFactors(n).count == 1
}

var primes: [Int] = []
var i = 0

while primes.count < 10001 {
    if isPrime(i) {
        primes.append(i)
    }

    i += 1
}

print(primes.last!)
